src/index.js

```diff
@@ -75,4 +102,28 @@ export default class SaucelabsConnector {
 
         await closeTunnel();
     }
+
+    async waitForFreeMachines (machinesCount, requestInterval, attemptsCount) {
+        var attempt          = 1;
+        var isMachinesEnough = false;
+
+        while (!isMachinesEnough && attempt <= attemptsCount) {
+            var freeMachinesCount = await this._getFreeMachinesCount();
+
+            isMachinesEnough = await freeMachinesCount >= machinesCount;
+
+            if (!isMachinesEnough) {
+                process.stdout.write('The amount of free machines is below than requested.\n');

```

Comment:
Let's log here how many available machines we have and how many machines we need